x/lockup/types/msgs_test.go

```diff
@@ -14,91 +14,246 @@ import (
-				msg.Coins = sdk.NewCoins(sdk.NewCoin("test1", sdk.NewInt(100000)), sdk.NewCoin("test2", sdk.NewInt(100000)))
-				return msg
-			}),
-			expectPass: false,
+			name: "invalid coins length",
+			msg: MsgBeginUnlocking{
+				Owner: addr1,
+				ID:    1,
+				Coins: sdk.NewCoins(sdk.NewCoin("test1", sdk.NewInt(100000)), sdk.NewCoin("test2", sdk.NewInt(100000))),
+			},
 		},
 		{
-			name: "zero token amount",
-			msg: createMsg(func(msg MsgLockTokens) MsgLockTokens {
-				msg.Coins = sdk.NewCoins(sdk.NewCoin("test1", sdk.NewInt(0)))
-				return msg
-			}),
-			expectPass: false,
+			name: "not positive coins amount",
+			msg: MsgBeginUnlocking{
+				Owner: addr1,
+				ID:    1,
+				Coins: sdk.NewCoins(sdk.NewCoin("test1", sdk.NewInt(0))),
+			},
 		},
 	}
 
 	for _, test := range tests {
-		if test.expectPass {
-			require.NoError(t, test.msg.ValidateBasic(), "test: %v", test.name)
-		} else {
-			require.Error(t, test.msg.ValidateBasic(), "test: %v", test.name)
-		}
+		t.Run(test.name, func(t *testing.T) {
+			if test.expectPass {
+				require.NoError(t, test.msg.ValidateBasic(), "test: %v", test.name)
+				require.Equal(t, test.msg.Route(), RouterKey)
+				require.Equal(t, test.msg.Type(), "begin_unlocking")
+				signers := test.msg.GetSigners()
+				require.Equal(t, len(signers), 1)
+				require.Equal(t, signers[0].String(), addr1)
+			} else {
+				require.Error(t, test.msg.ValidateBasic(), "test: %v", test.name)
+			}
+		})
+		
 	}
 }
 
-// TODO: Complete table driven tests for the remaining messages
+func TestMsgExtendLockup(t *testing.T) {
+	appParams.SetAddressPrefixes()
+	addr1, invalidAddr := generateTestAddrs()
 
-// MsgBeginUnlockingAll
+	tests := []struct {
+		name       string
+		msg        MsgExtendLockup
+		expectPass bool
+	}{
+		{
+			name: "proper msg",
+			msg: MsgExtendLockup{
+				Owner:    addr1,
+				ID:       1,
+				Duration: time.Hour,
+			},
+			expectPass: true,
+		},
+		{
+			name: "invalid owner",
+			msg: MsgExtendLockup{
+				Owner:    invalidAddr,
+				ID:       1,
+				Duration: time.Hour,
+			},
+		},
+		{
+			name: "invalid lockup ID",
+			msg: MsgExtendLockup{
+				Owner:    addr1,
+				ID:       0,
+				Duration: time.Hour,
+			},
+		},
+		{
+			name: "invalid duration",
+			msg: MsgExtendLockup{
+				Owner:    addr1,
+				ID:       1,
+				Duration: -1,
+			},
+		},
+	}
 
-// MsgBeginUnlocking
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if test.expectPass {
+				require.NoError(t, test.msg.ValidateBasic(), "test: %v", test.name)
+				require.Equal(t, test.msg.Route(), RouterKey)
+				require.Equal(t, test.msg.Type(), "edit_lockup")
+				signers := test.msg.GetSigners()
+				require.Equal(t, len(signers), 1)
+				require.Equal(t, signers[0].String(), addr1)
+			} else {
+				require.Error(t, test.msg.ValidateBasic(), "test: %v", test.name)
+			}
+		})
+	}
+}
 
-// MsgExtendLockup
+func generateTestAddrs() (string, string) {
```

Comment:
Can we move this helper to test utils?

Reply:
Yes I'll do it now

Reply:
What I meant is to define it on the app testing suite:
https://github.com/osmosis-labs/osmosis/blob/450e837158f4ae4cb908ec496da47fc169007218/app/apptesting/test_suite.go#L273

If you go with my other suggestion of [utilizing test suite](https://github.com/osmosis-labs/osmosis/pull/2205#discussion_r933548244), it would be convenient to reuse